### PR TITLE
Fix ncbi_datasets command line

### PR DESCRIPTION
**Pull request by mvdbeek**
multi-word options should be hyphenated.

FOR CONTRIBUTOR:
* [ ] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [ ] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

**Review by BradHolmes — APPROVED** [review]

**Comment by bgruening** [comment]
Tests are failing, unfortunately.

**Comment by mvdbeek** [comment]
Yep, looks like the results have changed it n the meantime.

**Comment by mvdbeek** [comment]
@BradHolmes let me know if that still looks good to you (https://github.com/galaxyproject/tools-iuc/pull/4502/commits/82ee3569bdb6cd7d9414e8547a440d370278097f dropped the sequence report from the outputs).
Alternatively we'd have to figure out why the sequence report disappeared, e.g for `datasets download genome accession 'GCF_000013305.1' 'GCF_000007445.1'  --exclude-gff3  --exclude-protein --exclude-rna --include-gtf --released-before '01/01/2007'`

File: tools/ncbi_datasets/datasets_genome.xml [hunk 1]
```diff
@@ -84,10 +84,6 @@ $filters.annotated
         <data name="genome_data_report" format="json" label="NCBI Genome Datasets: Data Report" from_work_dir="ncbi_dataset/data/assembly_data_report.jsonl">
             <filter>uncompressed</filter>
         </data>
-        <collection name="sequence_report" label="NCBI Genome Datasets: Sequence Data Report" type="list">
```
Comment (BradHolmes) [inline tools/ncbi_datasets/datasets_genome.xml:87]: This is still a valid collection for other assemblies, e.g. GCF_000001405.40

```
> datasets download genome accession GCF_000001405.40 --exclude-seq --exclude-rna --exclude-protein --exclude-genomic-cds --exclude-gff3
Downloading: ncbi_dataset.zip    7.37kB done
> unzip -l ncbi_dataset.zip 
Archive:  ncbi_dataset.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
     1604  05-02-2022 16:32   README.md
     2157  05-02-2022 16:32   ncbi_dataset/data/assembly_data_report.jsonl
    48833  05-02-2022 16:32   ncbi_dataset/data/GCF_000001405.40/sequence_report.jsonl
      317  05-02-2022 16:32   ncbi_dataset/data/dataset_catalog.json
```

Comment (mvdbeek) [reply tools/ncbi_datasets/datasets_genome.xml]: I've restored it in https://github.com/galaxyproject/tools-iuc/pull/4502/commits/0455bc367320897e5da6bd98b078abf58e3087a4 and added your example as a test in https://github.com/galaxyproject/tools-iuc/pull/4502/commits/0f1a4a806d70cc7379e915684e5e156e81c27703

**Review by BradHolmes — APPROVED** [review]
Looks good. Thanks!

File: tools/ncbi_datasets/datasets_genome.xml [hunk 2]
```diff
@@ -118,6 +118,26 @@ $filters.annotated
         </collection>
     </outputs>
     <tests>
+        <test expect_num_outputs="2">
+            <conditional name="query|subcommand">
```
Comment (bgruening) [inline tools/ncbi_datasets/datasets_genome.xml:122]: mh, that is a strange mixture of explicit vs. implicit naming. query is the outer section. But if it works ... maybe update with the next update.

**Pull request merged**